resources/blog/posts/2023/2023-08-16-rector-018-refocus-on-php.md

```diff
@@ -0,0 +1,52 @@
+---
+id: 50
+title: "Rector 0.18 - Refocus on PHP"
+perex: |
+    Before going to Rector 1.0, we need to refocus solely on PHP files. In this release, we're leaving a not-so-well-known feature that could handle some changes in configs and templates.
+---
+
+Like any other tools like ECS, PHPStan, PHP-CS-Fixer, PHPUnit, Pest, etc., Rector focuses on working with PHP files.
+Yet, in some cases, it could change YAML, TWIG, or files too. While using PHP classes in templates is a bad practice, it cannot be avoided like in Symfony configs.
+
+That's why I added a "special file processor" that could handle some non-php files too. This is a not-so-known feature that worked in quite a magic way.
+
+* those files are processed only if passed into the explicit paths, e.g.
+
+```bash
+# this could process non-PHP files
+bin/rector config src
+
+# this could not
+bin/rector src
+```
+
+* it only renames class names matching the FQN regex pattern
+
+```bash
+# This is skipped
+use App\SomeType;
+
+class: SomeType
+```
+
+The problem is Rector is built on AST to work with any form of PHP class naming - it's 100 % reliable and can handle whatever aliases.
+
+The magic mentioned above was hidden deep, leading to unexpected cases - some classes are renamed but are not.
+
+Instead, these files should be skipped entirely, and **Rector handles PHP files only**. This will show the same behavior as other CLI tools mention above and make you handle non-PHP files yourself in a consistent and aware way.
+
+## Potential for a Standalone tool
+
+What are the essential changes?
+
+* In Rector 0.17.3 we removed `NonPhpFileProcessor` and `NonPhpRectorInterface` - see [PR 4761](https://github.com/rectorphp/rector-src/pull/4761).
+
+* The `FileProcessorInterface` designed to support these magic changes is now deprecated and should be moved away from.
```

Comment:
I am thinking that `FileProcessorInterface` is fine to still exists, while marked as `@internal`

Reply:
Not possible, as that would still allow to change non-PHP files and lead to same result.
 Imagine PHPStan would have such interface, so it can analyse YAML files :)